docs/cartesian-product.adoc

```diff
@@ -387,23 +387,46 @@ class MyCartesianTestClass {
 }
 ----
 
-==== Requirements for the static factory method
+You can use an argument provider method in nested classes, as long as these are annotated with `@TestInstance(Lifecycle.PER_CLASS)`.
 
-There are multiple requirements the static factory method has to fulfill to qualify:
+[source,java]
+----
+class MyCartesianTestClass {
+	@Nested
+	@TestInstance(Lifecycle.PER_CLASS)
+	MyNestedCartesianTestClass {
```

Comment:
```suggestion
	class MyNestedCartesianTestClass {
```

Right?

Reply:
Right.

I did a rebase on main, and extracted the example to `CartesianTestExtensionDemo`, so these errors should now be caught..